### PR TITLE
Pruned Large Raw Transactions

### PR DESCRIPTION
**Pull request by jeremyfelder**
Added pruning of raw tx and sending only tx_id if pruned tx is >= 255KB
Also added logic for the backfilling nodes not to subscribe addresses from the queue

### CODE REVIEW

File: src/subscriptions.rs [hunk 1]
```diff
@@ -241,7 +281,20 @@ impl SubscriptionsHandler {
 
         match response {
             Ok(res) => debug!("Sent message with body '{}' and created message_id {}", msg_str, res.message_id.unwrap()),
-            Err(error) => debug!("SendMessageError: {:?}", error),
+            Err(error) => {
```
Comment (oleiba) [inline src/subscriptions.rs:291]: What does this pattern matching give?
It seems like the same handling before, only that `debug!` turned into `info!` + the handling of `Unknown` case a bit differently.

File: src/subscriptions.rs [hunk 2]
```diff
@@ -382,23 +435,25 @@ impl SubscriptionsManager {
         Ok(scripthashes)
     }
 
-    pub fn start(query: Arc<Query>, script_hash_comparison_status: Arc<AtomicBool>) -> SubscriptionsManager {
+    pub fn start(query: Arc<Query>, script_hash_comparison_status: Arc<AtomicBool>, backfill: bool) -> SubscriptionsManager {
```
Comment (oleiba) [inline src/subscriptions.rs:438]: Is there a case where this function would be called with `backfill` as false?

File: src/subscriptions.rs (hunk 1 above)
Comment (jeremyfelder) [reply src/subscriptions.rs:291]: Yes, expanded to handle the error for the unknown case so we dont see a raw buffer

File: src/subscriptions.rs [hunk 3]
```diff
@@ -221,10 +252,19 @@ impl SubscriptionsHandler {
                 warn!("notify_scripthash_subscriptions error - {}", raw_tx_res.err().unwrap());
                 return Ok(());
             }
+
+            let mut raw_tx = Value::Null;
+
+            let pruned_raw_tx = prune_raw_tx(raw_tx_res.unwrap());
+            if pruned_raw_tx.to_string().len() < 255000 {
```
Comment (oleiba) [inline src/subscriptions.rs]: Use a constant for 255000?

**Comment by oleiba** [comment]
Maybe we'd also want https://github.com/ZenGo-X/electrs/pull/14/files#diff-5cdcc3261dd77eadb37bcd60c5f85affcc9306454ae7bb8128e6d096ae51a631R399 to be `info!`?...

**Review by oleiba — APPROVED** [review]
Small changes.

File: src/subscriptions.rs (hunk 1 above)
Comment (oleiba) [reply src/subscriptions.rs:291]: Ok. So maybe just create a helper function to remove all the duplicate logs?
e.g. 
```
fn log_send_error(err_msg: &str) {
  info!("Error while sending message {}", err_msg)
} 
```
and have all the different branches to call it?...

**Pull request merged**